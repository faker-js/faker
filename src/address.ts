import type { Faker } from '.';
import type { Fake } from './fake';
import type { Helpers } from './helpers';

let f: Fake['fake'];

export class Address {
  readonly Helpers: Helpers;

  constructor(private readonly faker: Faker) {
    f = this.faker.fake;
    this.Helpers = this.faker.helpers;

    // Bind `this` so namespaced is working correctly
    for (const name of Object.getOwnPropertyNames(Address.prototype)) {
      if (name === 'constructor' || typeof this[name] !== 'function') {
        continue;
      }
      this[name] = this[name].bind(this);
    }

    // TODO @Shinigami92 2022-01-13: Need better strategy
    // @ts-expect-error
    this.direction.schema = {
      description:
        'Generates a direction. Use optional useAbbr bool to return abbreviation',
      sampleResults: ['Northwest', 'South', 'SW', 'E'],
    };
    // @ts-expect-error
    this.cardinalDirection.schema = {
      description:
        'Generates a cardinal direction. Use optional useAbbr boolean to return abbreviation',
      sampleResults: ['North', 'South', 'E', 'W'],
    };
    // @ts-expect-error
    this.ordinalDirection.schema = {
      description:
        'Generates an ordinal direction. Use optional useAbbr boolean to return abbreviation',
      sampleResults: ['Northwest', 'Southeast', 'SW', 'NE'],
    };
  }

  /**
   * Generates random zipcode from format. If format is not specified, the
   * locale's zip format is used.
   *
   * @method faker.address.zipCode
   * @param format
   */
  zipCode(format?: string) {
    // if zip format is not specified, use the zip format defined for the locale
    if (typeof format === 'undefined') {
      const localeFormat = this.faker.definitions.address.postcode;
      if (typeof localeFormat === 'string') {
        format = localeFormat;
      } else {
        format = this.faker.random.arrayElement(localeFormat);
      }
    }
    return this.Helpers.replaceSymbols(format);
  }

  /**
   * Generates random zipcode from state abbreviation. If state abbreviation is
   * not specified, a random zip code is generated according to the locale's zip format.
   * Only works for locales with postcode_by_state definition. If a locale does not
   * have a postcode_by_state definition, a random zip code is generated according
   * to the locale's zip format.
   *
   * @method faker.address.zipCodeByState
   * @param state
   */
  zipCodeByState(state: string) {
    const zipRange = this.faker.definitions.address.postcode_by_state[state];
    if (zipRange) {
      return this.faker.datatype.number(zipRange);
    }
    return this.faker.address.zipCode();
  }

  /**
   * Generates a random localized city name. The format string can contain any
   * method provided by faker wrapped in `{{}}`, e.g. `{{name.firstName}}` in
   * order to build the city name.
   *
   * If no format string is provided one of the following is randomly used:
   *
   * * `{{address.cityPrefix}} {{name.firstName}}{{address.citySuffix}}`
   * * `{{address.cityPrefix}} {{name.firstName}}`
   * * `{{name.firstName}}{{address.citySuffix}}`
   * * `{{name.lastName}}{{address.citySuffix}}`
   * * `{{address.cityName}}` when city name is available
   *
   * @method faker.address.city
   * @param format
   */
  city(format?: string | number) {
    const formats = [
      '{{address.cityPrefix}} {{name.firstName}}{{address.citySuffix}}',
      '{{address.cityPrefix}} {{name.firstName}}',
      '{{name.firstName}}{{address.citySuffix}}',
      '{{name.lastName}}{{address.citySuffix}}',
    ];

    if (!format && this.faker.definitions.address.city_name) {
      formats.push('{{address.cityName}}');
    }

    if (typeof format !== 'number') {
      format = this.faker.datatype.number(formats.length - 1);
    }

    return f(formats[format]);
  }

  /**
   * Return a random localized city prefix
   *
   * @method faker.address.cityPrefix
   */
  cityPrefix(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.address.city_prefix
    );
  }

  /**
   * Return a random localized city suffix
   *
   * @method faker.address.citySuffix
   */
  citySuffix(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.address.city_suffix
    );
  }

  /**
   * Returns a random city name
   *
   * @method faker.address.cityName
   */
  cityName(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.address.city_name
    );
  }

  /**
   * Returns a random localized street name
   *
   * @method faker.address.streetName
   */
  streetName(): string {
    let result: string;
    let suffix = this.faker.address.streetSuffix();
    if (suffix !== '') {
      suffix = ' ' + suffix;
    }

    switch (this.faker.datatype.number(1)) {
      case 0:
        result = this.faker.name.lastName() + suffix;
        break;
      case 1:
        result = this.faker.name.firstName() + suffix;
        break;
    }
    return result;
  }

  //
  // TODO: change all these methods that accept a boolean to instead accept an options hash.
  //

  /**
   * Returns a random localized street address
   *
   * @method faker.address.streetAddress
   * @param useFullAddress
   */
  streetAddress(useFullAddress: boolean = false): string {
    let address = '';
    switch (this.faker.datatype.number(2)) {
      case 0:
        address =
          this.Helpers.replaceSymbolWithNumber('#####') +
          ' ' +
          this.faker.address.streetName();
        break;
      case 1:
        address =
          this.Helpers.replaceSymbolWithNumber('####') +
          ' ' +
          this.faker.address.streetName();
        break;
      case 2:
        address =
          this.Helpers.replaceSymbolWithNumber('###') +
          ' ' +
          this.faker.address.streetName();
        break;
    }
    return useFullAddress
      ? address + ' ' + this.faker.address.secondaryAddress()
      : address;
  }

  /**
   * streetSuffix
   *
   * @method faker.address.streetSuffix
   */
  streetSuffix(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.address.street_suffix
    );
  }

  /**
   * streetPrefix
   *
   * @method faker.address.streetPrefix
   */
  streetPrefix(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.address.street_prefix
    );
  }

  /**
   * secondaryAddress
   *
   * @method faker.address.secondaryAddress
   */
  secondaryAddress() {
    return this.Helpers.replaceSymbolWithNumber(
      this.faker.random.arrayElement(['Apt. ###', 'Suite ###'])
    );
  }

  /**
   * county
   *
   * @method faker.address.county
   */
  county(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.address.county
    );
  }

  /**
   * country
   *
   * @method faker.address.country
   */
  country(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.address.country
    );
  }

  /**
   * countryCode
   *
   * @method faker.address.countryCode
   * @param alphaCode default alpha-2
   */
  countryCode(alphaCode: string = 'alpha-2'): string {
    if (alphaCode === 'alpha-2') {
      return this.faker.random.arrayElement(
        this.faker.definitions.address.country_code
      );
    }

    if (alphaCode === 'alpha-3') {
      return this.faker.random.arrayElement(
        this.faker.definitions.address.country_code_alpha_3
      );
    }

    return this.faker.random.arrayElement(
      this.faker.definitions.address.country_code
    );
  }

  /**
   * state
   *
   * @method faker.address.state
   * @param useAbbr
   */
  // TODO christopher 2022-01-13: useAbbr not in use
  state(useAbbr?: boolean): string {
    return this.faker.random.arrayElement(this.faker.definitions.address.state);
  }

  /**
   * stateAbbr
   *
   * @method faker.address.stateAbbr
   */
  stateAbbr(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.address.state_abbr
    );
  }

  /**
   * latitude
   *
   * @method faker.address.latitude
   * @param max default is 90
   * @param min default is -90
   * @param precision default is 4
   */
  latitude(max: number = 90, min: number = -90, precision: number = 4): string {
    return this.faker.datatype
      .number({
        max: max,
        min: min,
        precision: parseFloat((0.0).toPrecision(precision) + '1'),
      })
      .toFixed(precision);
  }

  /**
   * longitude
   *
   * @method faker.address.longitude
   * @param max default is 180
   * @param min default is -180
   * @param precision default is 4
   */
  longitude(
    max: number = 180,
    min: number = -180,
    precision: number = 4
  ): string {
    return this.faker.datatype
      .number({
        max: max,
        min: min,
        precision: parseFloat((0.0).toPrecision(precision) + '1'),
      })
      .toFixed(precision);
  }

  /**
   *  direction
   *
   * @method faker.address.direction
   * @param useAbbr return direction abbreviation. defaults to false
   */
  direction(useAbbr: boolean = false) {
    if (!useAbbr) {
      return this.faker.random.arrayElement(
        this.faker.definitions.address.direction
      );
    }
    return this.faker.random.arrayElement(
      this.faker.definitions.address.direction_abbr
    );
  }

  /**
   * cardinal direction
   *
   * @method faker.address.cardinalDirection
   * @param useAbbr return direction abbreviation. defaults to false
   */
  cardinalDirection(useAbbr: boolean = false): string {
    if (!useAbbr) {
      return this.faker.random.arrayElement(
        this.faker.definitions.address.direction.slice(0, 4)
      );
    }
    return this.faker.random.arrayElement(
      this.faker.definitions.address.direction_abbr.slice(0, 4)
    );
  }

  /**
   * ordinal direction
   *
   * @method faker.address.ordinalDirection
   * @param useAbbr return direction abbreviation. defaults to false
   */
  ordinalDirection(useAbbr: boolean = false): string {
    if (!useAbbr) {
      return this.faker.random.arrayElement(
        this.faker.definitions.address.direction.slice(4, 8)
      );
    }
    return this.faker.random.arrayElement(
      this.faker.definitions.address.direction_abbr.slice(4, 8)
    );
  }

  nearbyGPSCoordinate(
    coordinate?: number[],
    radius?: number,
    isMetric?: boolean
  ): string[] {
    function randomFloat(min: number, max: number): number {
      return Math.random() * (max - min) + min;
    }
    function degreesToRadians(degrees: number): number {
      return degrees * (Math.PI / 180.0);
    }
    function radiansToDegrees(radians: number): number {
      return radians * (180.0 / Math.PI);
    }
    function kilometersToMiles(miles: number): number {
      return miles * 0.621371;
    }
    function coordinateWithOffset(
      coordinate: number[],
      bearing: number,
      distance: number,
      isMetric: boolean
    ): number[] {
      const R = 6378.137; // Radius of the Earth (http://nssdc.gsfc.nasa.gov/planetary/factsheet/earthfact.html)
      const d = isMetric ? distance : kilometersToMiles(distance); // Distance in km

      const lat1 = degreesToRadians(coordinate[0]); //Current lat point converted to radians
      const lon1 = degreesToRadians(coordinate[1]); //Current long point converted to radians

      const lat2 = Math.asin(
        Math.sin(lat1) * Math.cos(d / R) +
          Math.cos(lat1) * Math.sin(d / R) * Math.cos(bearing)
      );

      let lon2 =
        lon1 +
        Math.atan2(
          Math.sin(bearing) * Math.sin(d / R) * Math.cos(lat1),
          Math.cos(d / R) - Math.sin(lat1) * Math.sin(lat2)
        );

      // Keep longitude in range [-180, 180]
      if (lon2 > degreesToRadians(180)) {
        lon2 = lon2 - degreesToRadians(360);
      } else if (lon2 < degreesToRadians(-180)) {
        lon2 = lon2 + degreesToRadians(360);
      }

      return [radiansToDegrees(lat2), radiansToDegrees(lon2)];
    }

    // If there is no coordinate, the best we can do is return a random GPS coordinate.
    if (coordinate === undefined) {
      return [this.faker.address.latitude(), this.faker.address.longitude()];
    }
    radius ||= 10.0;
    isMetric ||= false;

    // TODO: implement either a gaussian/uniform distribution of points in circular region.
    // Possibly include param to function that allows user to choose between distributions.

    // This approach will likely result in a higher density of points near the center.
    const randomCoord = coordinateWithOffset(
      coordinate,
      degreesToRadians(Math.random() * 360.0),
      radius,
      isMetric
    );
    return [randomCoord[0].toFixed(4), randomCoord[1].toFixed(4)];
  }

  /**
   * Return a random time zone
   *
   * @method faker.address.timeZone
   */
  timeZone(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.address.time_zone
    );
  }
}
