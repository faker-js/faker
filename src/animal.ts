import type { Faker } from '.';

export class Animal {
  constructor(private readonly faker: Faker) {
    // Bind `this` so namespaced is working correctly
    for (const name of Object.getOwnPropertyNames(Animal.prototype)) {
      if (name === 'constructor' || typeof this[name] !== 'function') {
        continue;
      }
      this[name] = this[name].bind(this);
    }
  }

  /**
   * Returns a random dog breed.
   *
   */
  dog(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.dog);
  }

  /**
   * Returns a random cat breed.
   *
   */
  cat(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.cat);
  }

  /**
   * Returns a random snake species.
   *
   */
  snake(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.snake);
  }

  /**
   * Returns a random bear species.
   *
   */
  bear(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.bear);
  }

  /**
   * Returns a random lion species.
   *
   */
  lion(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.lion);
  }

  /**
   * Returns a random cetacean species.
   *
   * @method faker.animal.cetacean
   */
  cetacean(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.animal.cetacean
    );
  }

  /**
   * Returns a random horse breed.
   *
   */
  horse(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.horse);
  }

  /**
   * Returns a random bird species.
   *
   */
  bird(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.bird);
  }

  /**
   * Returns a random cow species.
   *
   */
  cow(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.cow);
  }

  /**
   * Returns a random fish species.
   *
   */
  fish(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.fish);
  }

  /**
   * Returns a random crocodilian species.
   *
   */
  crocodilia(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.animal.crocodilia
    );
  }

  /**
   * Returns a random insect species.
   *
   */
  insect(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.insect);
  }

  /**
   * Returns a random rabbit species
   *
   */
  rabbit(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.rabbit);
  }

  /**
   * Returns a random animal type.
   *
   */
  type(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.type);
  }
}
