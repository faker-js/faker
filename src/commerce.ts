import type { Faker } from '.';
import { deprecated } from './internal/deprecated';

/**
 * Module to generate commerce and product related entries.
 */
export class Commerce {
  constructor(private readonly faker: Faker) {
    // Bind `this` so namespaced is working correctly
    for (const name of Object.getOwnPropertyNames(Commerce.prototype)) {
      if (name === 'constructor' || typeof this[name] !== 'function') {
        continue;
      }
      this[name] = this[name].bind(this);
    }
  }

  /**
   * Returns a human readable color name.
   *
   * @example
   * faker.commerce.color() // 'red'
   *
   * @deprecated
   */
  color(): string {
    deprecated({
      deprecated: 'faker.commerce.color()',
      proposed: 'faker.color.human()',
      since: 'v7.0.0',
      until: 'v8.0.0',
    });
    return this.faker.color.human();
  }

  /**
   * Returns a department inside a shop.
   *
   * @example
   * faker.commerce.department() // 'Garden'
   */
  department(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.commerce.department
    );
  }

  /**
   * Return a percentage value in decimal format between 0 and 1
   * with percision of two decimal place.
   *
   * @param precision Precision of the generated number. Defaults to `0.01`.
   *
   * @example
   * this.getPercentage(0.1) // 0.4
   * this.getPercentage(0.01) // 0.36
   */
  percentage(precision: number): number {
    return this.faker.datatype.float({
      min: 0,
      max: 1,
      precision: precision || 0.01,
    });
  }

  /**
   * Generates a random descriptive product name.
   *
   * @example
   * faker.commerce.productName() // 'Incredible Soft Gloves'
   */
  productName(): string {
    return (
      this.faker.commerce.productAdjective() +
      ' ' +
      this.faker.commerce.productMaterial() +
      ' ' +
      this.faker.commerce.product()
    );
  }

  /**
   * Generates a price between min and max (inclusive).
   *
   * @param min The minimum price. Defaults to `1`.
   * @param max The maximum price. Defaults to `1000`.
   * @param dec The number of decimal places. Defaults to `2`.
   * @param symbol The currency value to use. Defaults to `''`.
   *
   * @example
   * faker.commerce.price() // 828.00
   * faker.commerce.price(100) // 904.00
   * faker.commerce.price(100, 200) // 154.00
   * faker.commerce.price(100, 200, 0) // 133
   * faker.commerce.price(100, 200, 0, '$') // $114
   */
  price(
    min: number = 1,
    max: number = 1000,
    dec: number = 2,
    symbol: string = ''
  ): string {
    if (min < 0 || max < 0) {
      return `${symbol}${0.0}`;
    }

    const randValue = this.faker.datatype.number({ max: max, min: min });

    return (
      symbol +
      (Math.round(randValue * Math.pow(10, dec)) / Math.pow(10, dec)).toFixed(
        dec
      )
    );
  }

  /**
   * Returns an adjective describing a product.
   *
   * @example
   * faker.commerce.productAdjective() // 'Handcrafted'
   */
  productAdjective(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.commerce.product_name.adjective
    );
  }

  /**
   * Returns a material of a product.
   *
   * @example
   * faker.commerce.productMaterial() // 'Rubber'
   */
  productMaterial(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.commerce.product_name.material
    );
  }

  /**
   * Returns a short product name.
   *
   * @example
   * faker.commerce.product() // 'Computer'
   */
  product(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.commerce.product_name.product
    );
  }

  /**
   * Returns a product description.
   *
   * @example
   * faker.commerce.productDescription() // 'Andy shoes are designed to keeping...'
   */
  productDescription(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.commerce.product_description
    );
  }
}
