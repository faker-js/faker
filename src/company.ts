import type { Faker } from '.';
import type { Fake } from './fake';

let f: Fake['fake'];

export class Company {
  constructor(private readonly faker: Faker) {
    f = this.faker.fake;

    // Bind `this` so namespaced is working correctly
    for (const name of Object.getOwnPropertyNames(Company.prototype)) {
      if (name === 'constructor' || typeof this[name] !== 'function') {
        continue;
      }
      this[name] = this[name].bind(this);
    }
  }

  /**
   * suffixes
   *
   * @method faker.company.suffixes
   */
  suffixes(): string[] {
    // Don't want the source array exposed to modification, so return a copy
    return this.faker.definitions.company.suffix.slice(0);
  }

  /**
   * companyName
   *
   * @method faker.company.companyName
   * @param format
   */
  companyName(format?: number): string {
    const formats = [
      '{{name.lastName}} {{company.companySuffix}}',
      '{{name.lastName}} - {{name.lastName}}',
      '{{name.lastName}}, {{name.lastName}} and {{name.lastName}}',
    ];

    if (typeof format !== 'number') {
      format = this.faker.datatype.number(formats.length - 1);
    }

    return f(formats[format]);
  }

  /**
   * companySuffix
   *
   * @method faker.company.companySuffix
   */
  companySuffix(): string {
    return this.faker.random.arrayElement(this.faker.company.suffixes());
  }

  /**
   * catchPhrase
   *
   * @method faker.company.catchPhrase
   */
  catchPhrase(): string {
    return f(
      '{{company.catchPhraseAdjective}} {{company.catchPhraseDescriptor}} {{company.catchPhraseNoun}}'
    );
  }

  /**
   * bs
   *
   * @method faker.company.bs
   */
  bs(): string {
    return f('{{company.bsBuzz}} {{company.bsAdjective}} {{company.bsNoun}}');
  }

  /**
   * catchPhraseAdjective
   *
   * @method faker.company.catchPhraseAdjective
   */
  catchPhraseAdjective(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.company.adjective
    );
  }

  /**
   * catchPhraseDescriptor
   *
   * @method faker.company.catchPhraseDescriptor
   */
  catchPhraseDescriptor(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.company.descriptor
    );
  }

  /**
   * catchPhraseNoun
   *
   * @method faker.company.catchPhraseNoun
   */
  catchPhraseNoun(): string {
    return this.faker.random.arrayElement(this.faker.definitions.company.noun);
  }

  /**
   * bsAdjective
   *
   * @method faker.company.bsAdjective
   */
  bsAdjective(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.company.bs_adjective
    );
  }

  /**
   * bsBuzz
   *
   * @method faker.company.bsBuzz
   */
  bsBuzz(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.company.bs_verb
    );
  }

  /**
   * bsNoun
   *
   * @method faker.company.bsNoun
   */
  bsNoun(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.company.bs_noun
    );
  }
}
