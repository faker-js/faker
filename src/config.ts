/**
 * The possible configuration options, that can be set.
 * This type exists to be extended for plugins via type augmentation.
 *
 * The `@default` tag is used to indicate the default value, that should be used if, the config is absent.
 */
export interface FakerConfig {
  /**
   * The function used to generate the `refDate` date instance, if not provided as method param.
   * The function must return a new valid `Date` instance for every call.
   *
   * @see [Reproducible Results](https://fakerjs.dev/guide/usage.html#reproducible-results)
   * @see faker.seed(): For generating reproducible values.
   *
   * @since 9.0.0
   *
   * @default () => new Date()
   */
  defaultRefDate?: () => Date;
}
