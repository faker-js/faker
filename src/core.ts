import type { FakerConfig } from './config';
import type { LocaleDefinition } from './definitions';
import type { Randomizer } from './randomizer';

/**
 * The core grants access to the locale data, the randomizer and config settings.
 */
export interface FakerCore {
  /**
   * The locale data associated with this instance.
   *
   * Always present, but it might be empty if the locale data is not available.
   */
  readonly locale: LocaleDefinition;

  /**
   * The randomizer used to generate random values.
   */
  readonly randomizer: Randomizer;

  /**
   * The configuration settings used by this instance.
   */
  readonly config: FakerConfig;
}
