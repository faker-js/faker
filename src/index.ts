import { Faker } from './faker';
import allLocales from './locales';

export type {
  AnimalDefinitions,
  BedDefinitions,
  BeerDefinitions,
  BookcaseDefinitions,
  BookDefinitions,
  ChairDefinitions,
  CigarDefinitions,
  CigaretteDefinitions,
  ClothingDefinitions,
  ColorDefinitions,
  CommerceDefinitions,
  CommerceProductNameDefinitions,
  CompanyDefinitions,
  ComputerDefinitions,
  CondomDefinitions,
  CosmeticDefinitions,
  DatabaseDefinitions,
  DateDefinitions,
  DateEntryDefinition,
  DrinkDefinitions,
  FabricDefinitions,
  FinanceCurrencyEntryDefinitions,
  FinanceDefinitions,
  FishDefinitions,
  FlightsDefinitions,
  FlowerDefinitions,
  FoodDefinitions,
  FurnitureDefinitions,
  GlassesDefinitions,
  HackerDefinitions,
  HatDefinitions,
  InternetDefinitions,
  JewelsDefinitions,
  KitchenDefinitions,
  LighterDefinitions,
  LiquorDefinitions,
  LocaleDefinition,
  LocationDefinitions,
  LoremDefinitions,
  MeatDefinitions,
  MedicationDefinitions,
  MonumentDefinitions,
  MovieDefinitions,
  MusicDefinitions,
  ParfumDefinitions,
  PersonDefinitions,
  PersonTitleDefinitions,
  PhoneDefinitions,
  PlantDefinitions,
  PrinterDefinitions,
  RecipeDefinitions,
  ScienceDefinitions,
  ShoesDefinitions,
  SmokingFilterDefinitions,
  SmokingRollingPaperDefinitions,
  SofaDefinitions,
  StaysDefinitions,
  SystemDefinitions,
  SystemMimeTypeEntryDefinitions,
  TableDefinitions,
  TattooDefinitions,
  TireDefinitions,
  TobaccoDefinitions,
  ToyDefinitions,
  TvDefinitions,
  VehicleDefinitions,
  VideogameDefinitions,
  WardrobeDefinitions,
  WatchDefinitions,
  WheelDefinitions,
  WineDefinitions,
  WordDefinitions,
} from './definitions';
export { FakerError } from './errors/faker-error';
export type { FakerOptions, UsableLocale, UsedLocales } from './faker';
export type { AnimalModule } from './modules/animal';
export type { BedModule } from './modules/bed';
export type { BeerModule } from './modules/beer';
export type { BookModule } from './modules/book';
export type { BookcaseModule } from './modules/bookcase';
export type { ChairModule } from './modules/chair';
export type { ClothingModule } from './modules/clothing';
export type {
  Casing,
  ColorFormat,
  ColorModule,
  CSSFunction,
  CSSSpace,
  NumberColorFormat,
  StringColorFormat,
} from './modules/color';
export type { CommerceModule } from './modules/commerce';
export type { CompanyModule } from './modules/company';
export type { ComputerModule } from './modules/computer';
export type { CondomModule } from './modules/condom';
export type { CosmeticModule } from './modules/cosmetic';
export type { DatabaseModule } from './modules/database';
export type { DatatypeModule } from './modules/datatype';
export type { DateModule } from './modules/date';
export type { DrinkModule } from './modules/drink';
export type { FabricModule } from './modules/fabric';
export type { FinanceModule } from './modules/finance';
export type { FishModule } from './modules/fish';
export type { FlightsModule } from './modules/flights';
export type { FlowerModule } from './modules/flower';
export type { FoodModule } from './modules/food';
export type { FurnitureModule } from './modules/furniture';
export type { GitModule } from './modules/git';
export type { GlassesModule } from './modules/glasses';
export type { HackerModule } from './modules/hacker';
export type { HatModule } from './modules/hat';
export type { HelpersModule } from './modules/helpers';
export type { ImageModule } from './modules/image';
export type { InternetModule } from './modules/internet';
export type { JewelsModule } from './modules/jewels';
export type { KitchenModule } from './modules/kitchen';
export type { LighterModule } from './modules/lighter';
export type { LiquorModule } from './modules/liquor';
export type { LocationModule } from './modules/location';
export type { LoremModule } from './modules/lorem';
export type { MeatModule } from './modules/meat';
export type { MedicationModule } from './modules/medication';
export type { MonumentModule } from './modules/monument';
export type { MovieModule } from './modules/movie';
export type { MusicModule } from './modules/music';
export type { ParfumModule } from './modules/parfum';
export { Sex } from './modules/person';
export type { PersonModule, SexType } from './modules/person';
export type { PhoneModule } from './modules/phone';
export type { PlantModule } from './modules/plant';
export type { RandomModule } from './modules/random';
export type { RecipeModule } from './modules/recipe';
export type { ChemicalElement, ScienceModule, Unit } from './modules/science';
export type { ShoesModule } from './modules/shoes';
export type { SmokingFilterModule } from './modules/smokingFilter';
export type { SmokingRollingPaperModule } from './modules/smokingRollingPaper';
export type { SofaModule } from './modules/sofa';
export type { StaysModule } from './modules/stays';
export type { StringModule } from './modules/string';
export type { SystemModule } from './modules/system';
export type { TableModule } from './modules/table';
export type { TattooModule } from './modules/tattoo';
export type { TireModule } from './modules/tire';
export type { TobaccoModule } from './modules/tobacco';
export type { ToyModule } from './modules/toy';
export type { TvModule } from './modules/tv';
export type { VehicleModule } from './modules/vehicle';
export type { VideogameModule } from './modules/videogame';
export type { WardrobeModule } from './modules/wardrobe';
export type { WatchModule } from './modules/watch';
export type { WheelModule } from './modules/wheel';
export type { WineModule } from './modules/wine';
export type { WordModule } from './modules/word';
export { Faker };

// since we are requiring the top level of faker, load all locales by default
export const faker: Faker = new Faker({
  locales: allLocales,
});
