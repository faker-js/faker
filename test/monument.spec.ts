import { beforeEach, describe, expect, it } from 'vitest';
import { faker } from '../src';
import { seededTests } from './support/seededRuns';

const NON_SEEDED_BASED_RUN = 5;

describe('monument', () => {
  beforeEach(() => {
    faker.locale = 'en';
  });

  seededTests(faker, 'monument', (t) => {
    t.itEach(
      'monumentName',
      'monumentDescription',
      'monumentLocation',
      'monumentCountry',
      'monumentYear',
      'monumentDateEstabilished'
    );
  });

  describe(`random seeded tests for seed ${faker.seed()}`, () => {
    for (let i = 1; i <= NON_SEEDED_BASED_RUN; i++) {
      describe('name()', () => {
        it('should return a name', () => {
          const name = faker.monument.monumentName();

          expect(name).toBeTruthy();
          expect(name).toBeTypeOf('string');
          expect(faker.definitions.monument?.name).toContain(name);
        });
      });

      describe('description()', () => {
        it('should return a description', () => {
          const description = faker.monument.monumentDescription();

          expect(description).toBeTruthy();
          expect(description).toBeTypeOf('string');
          expect(faker.definitions.monument?.description).toContain(
            description
          );
        });
      });

      describe('location()', () => {
        it('should return a location', () => {
          const location = faker.monument.monumentLocation();

          expect(location).toBeTruthy();
          expect(location).toBeTypeOf('string');
        });
      });

      describe('country()', () => {
        it('should return a country', () => {
          const country = faker.monument.monumentCountry();

          expect(country).toBeTruthy();
          expect(country).toBeTypeOf('string');
        });
      });

      describe('year()', () => {
        it('should return a brand', () => {
          const year = faker.monument.monumentYear();

          expect(year).toBeTruthy();
          expect(year).toBeTypeOf('number');
        });
      });

      describe('dateEstabilished()', () => {
        it('should return a date estabilished', () => {
          const dateEstabilished = faker.monument.monumentDateEstabilished();

          expect(dateEstabilished).toBeTruthy();
        });
      });
    }
  });
});
