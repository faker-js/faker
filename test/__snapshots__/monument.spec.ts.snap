// Vitest Snapshot v1

exports[`monument > 42 > monumentCountry 1`] = `"Haiti"`;

exports[`monument > 42 > monumentDateEstabilished 1`] = `1857-01-31T04:06:40.302Z`;

exports[`monument > 42 > monumentDescription 1`] = `"Soliders in the US Army's 10th Mountain Division used Camp Hale to learn skiing, climbing, and snowshoeing to fight in the Italian Alps during World War II. Part of White River National Forest."`;

exports[`monument > 42 > monumentLocation 1`] = `"Port Valentine"`;

exports[`monument > 42 > monumentName 1`] = `"Florissant Fossil Beds"`;

exports[`monument > 42 > monumentYear 1`] = `1856`;

exports[`monument > 1211 > monumentCountry 1`] = `"Uganda"`;

exports[`monument > 1211 > monumentDateEstabilished 1`] = `1940-03-07T12:31:18.349Z`;

exports[`monument > 1211 > monumentDescription 1`] = `"This monument preserves three prehistoric sites with 206 prehistoric mounds, notable for 31 unusual mounds in the shape of mammals, birds, or reptiles."`;

exports[`monument > 1211 > monumentLocation 1`] = `"La Crosse"`;

exports[`monument > 1211 > monumentName 1`] = `"Mill Springs Battlefield"`;

exports[`monument > 1211 > monumentYear 1`] = `1940`;

exports[`monument > 1337 > monumentCountry 1`] = `"Equatorial Guinea"`;

exports[`monument > 1337 > monumentDateEstabilished 1`] = `1840-03-16T21:36:43.823Z`;

exports[`monument > 1337 > monumentDescription 1`] = `"This site preserves the A. G. Gaston Motel, which was bombed by the KKK after Martin Luther King Jr. and Birmingham campaign leaders had stayed there, inciting the Birmingham riot of 1963"`;

exports[`monument > 1337 > monumentLocation 1`] = `"New Carmella"`;

exports[`monument > 1337 > monumentName 1`] = `"Chiricahua"`;

exports[`monument > 1337 > monumentYear 1`] = `1839`;
